src/Modal.js

```diff
@@ -277,6 +277,11 @@ const Modal = React.createClass({
     }
   },
 
+  componentWillUnmount() {
+    this.onHide();
+    this.props.onHide();
```

Comment:
I don't think it makes sense to call `props.onHide` there. We have `props.onHide` because you can't control when the user will interact with the `Modal` in such a way as to want to close it. However, you (presumably) have control over when you are unmounting the `Modal`, so you have the opportunity to take whatever actions are relevant there. If you're unmounting the `Modal` for a reason that is similar to the user initiating a "hide" action, then you can invoke that callback yourself. If you're unmounting the `Modal` for some different reason, then I don't want to force that behavior upon you.